### ISSUE TITLE
Turn some dependencies into core services

### ISSUE THREAD

**Issue by JahlomP** [pencilblue/pencilblue#852]
What if we could write some of the dependencies as services. Our dependencies will reduce and we will have total control over them. 

Additions of new features will be done fast. We wouldn't wait for that module developer to update the module before we have it. We could just add it to our own service.  
We will also add what we need to reduce code size. Users won't have to install a lot of dependencies.

Some include htmlencode, node-uuid, node.extend, node-process, locale and sanitize-html.

**Comment by JahlomP** [comment]
With the node introducing two paths of node, not depending on a lot of external modules will reduce the uncertainty of a module supporting a node version or not. We will have full control over major parts of the platform.

**Comment by brianhyder** [comment]
There is a fine balance that has to be made. Too many dependencies and
finding matching versions between packages gets harder. Writing pieces of
code that are not your core competency means that you are maintaining lots
of code you don't know is efficient.

Think of pencilblue like a car. It is made of a lot of parts. Some we made
ourselves and some we didn't. If we had to make all the parts ourselves we
would never finish assembling the car and worse it wouldn't perform because
we tried to make something we just don't have a lot of knowledge around.

Hopefully, that makes sense.
